libdnf/sack/query.cpp

```diff
@@ -112,8 +130,24 @@ NevraID::parse(Pool * pool, const char * nevraPattern)
     if (!(name = pool_strn2id(pool, nevraPattern, nameLen, 0)))
         return false;
     ++evrDelim;
-    if (!(evr = pool_strn2id(pool, evrDelim, archDelim - evrDelim, 0)))
-        return false;
+
+    // evr
+    if (createEVRId) {
+        // strip epoch "0:"
+        if (evrDelim[0] == '0' && evrDelim[1] == ':') {
+            evrDelim += 2;
+        }
+        if (!(evr = pool_strn2id(pool, evrDelim, archDelim - evrDelim, 0))) {
+            return false;
+        }
+    } else {
+        size_t evr_size = archDelim - evrDelim;
+        if (evr_size == 0) {
```

Comment:
`evr_size == 0` is always false, isn't it?
What about to replace 5 lines with one: `evr_str = std::string(evrDelim, archDelim);`